File: binarySearch/binarySearch.py


def binary_search(nums: list, target: int) -> int:
    left, right = 0, len(nums)
    while left < right:
        middle = left + (right - left) >> 1
        if nums[middle] < target:
            left = middle + 1
        elif nums[middle] > target:
            right = middle
        else:
            return middle
    return

def main():
    test_list = [1,2,3,4,5,6,7,8]
    target = 3
    print(test_list, target)
    print(binary_search(test_list, target))

if __name__ == '__main__':
    main()

File: binarySearch/binarySearch.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func generateRandomArray(maxSize int, maxValue int) []int {
	rand.Seed(time.Now().UnixNano())
	arr := []int{}
	for i := 0; i < int(rand.Float32()*float32(maxSize+1)); i++ {
		arr = append(arr, int(rand.Float32()*float32(maxValue+1))-int(rand.Float32()*float32(maxValue+1)))
	}
	return arr
}

func binarySearch(arr []int) int {
	left, right := 0, len(arr)-1
	mid := left + (right-left)>>1
	fmt.Println(left, mid, right)
	return 0
}

func main() {
	testTime := 10
	maxSize := 10
	maxValue := 100
	for i := 0; i < testTime; i++ {

		arr := generateRandomArray(maxSize, maxValue)
		fmt.Println(arr)
		binarySearch(arr)
		break
	}
}
